Python/bytecodes.c

```diff
@@ -1158,31 +1169,43 @@ dummy_func(
             }
         }
 
-        inst(LOAD_NAME, ( -- v)) {
-            PyObject *name = GETITEM(frame->f_code->co_names, oparg);
-            PyObject *locals = LOCALS();
-            if (locals == NULL) {
+        op(_LOAD_NAME_INTRO, (-- mod_or_class_dict, name)) {
+            name = GETITEM(frame->f_code->co_names, oparg);
+            mod_or_class_dict = LOCALS();
+            if (mod_or_class_dict == NULL) {
                 _PyErr_Format(tstate, PyExc_SystemError,
                               "no locals when loading %R", name);
                 goto error;
             }
-            if (PyDict_CheckExact(locals)) {
-                v = PyDict_GetItemWithError(locals, name);
+            Py_INCREF(mod_or_class_dict);
+        }
+
+        op(_LOAD_CLASSDICT_OR_GLOBAL_INTRO, (mod_or_class_dict -- mod_or_class_dict, name)) {
+            name = GETITEM(frame->f_code->co_names, oparg);
+        }
+
+        op(_LOAD_NAME_COMMON, (mod_or_class_dict, name -- v)) {
```

Comment:
As above, the name should describe what it does, not how it is (currently) used.
This works for any dict-like object on TOS.

Maybe name it `LOAD_FROM_DICT_OR_GLOBALS`, or something like that.

Reply:
Renamed to _LOAD_FROM_DICT_OR_GLOBALS (Brandt told me the convention for the "op" is a leading underscore).